zeebe/backup/src/test/java/io/camunda/zeebe/backup/management/BackupServiceImplTest.java

```diff
@@ -417,7 +417,49 @@ void shouldListAvailableBackups() {
         .thenReturn(CompletableFuture.completedFuture(List.of(backup1, backup2)));
 
     // when
-    final var backups = backupService.listBackups(partitionId, concurrencyControl).join();
+    final var backups = backupService.listBackups(partitionId, null, concurrencyControl).join();
+
+    // then
+    assertThat(backups).containsExactlyInAnyOrder(backup1, backup2);
+  }
+
+  @Test
+  void shouldListAvailableBackupsWithPrefix() {
+    // given
+    final int partitionId = 2;
+    final var backup1 = mock(BackupStatus.class);
+    final var backup2 = mock(BackupStatus.class);
+    when(backup1.id()).thenReturn(new BackupIdentifierImpl(1, partitionId, 11));
+    when(backup2.id()).thenReturn(new BackupIdentifierImpl(1, partitionId, 21));
+
+    when(backupStore.list(
+            new BackupIdentifierWildcardImpl(
+                Optional.empty(), Optional.of(partitionId), Optional.empty())))
+        .thenReturn(CompletableFuture.completedFuture(List.of(backup1, backup2)));
+
+    // when
+    final var backups = backupService.listBackups(partitionId, "1*", concurrencyControl).join();
+
+    // then
+    assertThat(backups).singleElement().isEqualTo(backup1);
+  }
+
+  @Test
+  void shouldListAllAvailableBackupsWithEmptyPrefix() {
+    // given
+    final int partitionId = 2;
+    final var backup1 = mock(BackupStatus.class);
+    final var backup2 = mock(BackupStatus.class);
+    when(backup1.id()).thenReturn(new BackupIdentifierImpl(1, partitionId, 11));
+    when(backup2.id()).thenReturn(new BackupIdentifierImpl(1, partitionId, 21));
+
+    when(backupStore.list(
+            new BackupIdentifierWildcardImpl(
+                Optional.empty(), Optional.of(partitionId), Optional.empty())))
+        .thenReturn(CompletableFuture.completedFuture(List.of(backup1, backup2)));
+
+    // when
+    final var backups = backupService.listBackups(partitionId, "*", concurrencyControl).join();
```

Comment:
❓ Actually here we now us the wildcard, but in the test method we say empty prefix. Do we want to test something different here?

Reply:
Test was removed again later on anyway.